avocado/core/job.py

```diff
@@ -462,6 +465,7 @@ def _run(self):
         """
         self._setup_job_results()
         self.__start_job_logging()
+        self.job_pre_post_dispatcher.map_methods('pre', self)
```

Comment:
Yep, trello card about it:

https://trello.com/c/w4bSnrRi/665-pre-job-plugins-re-evaluate-when-they-re-called